src/charm.py

```diff
@@ -158,17 +154,14 @@ def _generate_root_certificate(self) -> None:
         """
         if not self._config_ca_common_name:
             raise ValueError("CA common name should not be empty")
-        private_key_password = generate_password()
-        private_key = generate_private_key(password=private_key_password.encode())
+        private_key = generate_private_key()
         ca_certificate = generate_ca(
             private_key=private_key,
             subject=self._config_ca_common_name,
-            private_key_password=private_key_password.encode(),
         )
         secret_content = {
-            "private-key-password": private_key_password,
-            "private-key": private_key.decode(),
-            "ca-certificate": ca_certificate.decode(),
+            "private-key": private_key,
+            "ca-certificate": ca_certificate,
         }
         if self._root_certificate_is_stored:
             secret = self.model.get_secret(label=CA_CERTIFICATES_SECRET_LABEL)
```

Comment:
I know this wasn't done in this PR but the CA secret should expire based on `_config_root_ca_certificate_validity` not on `_config_certificate_validity`

Reply:
I agree but this is unrelated to this change, I suggest keeping this task focused on this refactoring and not include other bug fixes

Reply:
Ticket created #214